web/modules/custom/rdf_entity/src/Entity/RdfEntitySparqlStorage.php

```diff
@@ -656,34 +659,64 @@ protected function doSave($id, EntityInterface $entity) {
     $target_graph = $this->getGraphHandler()->getTargetGraphFromEntity($entity);
     $graph_uri = $this->getBundleGraphUri($bundle, $target_graph);
 
-    $insert = '';
+    $graph = new Graph($graph_uri);
+
     $properties = $this->mappingHandler->getEntityTypeMappedProperties($entity);
-    $subj = '<' . (string) $id . '>';
     $properties_list = "<" . implode(">, <", $properties['flat']) . ">";
     foreach ($entity->toArray() as $field_name => $field) {
       foreach ($field as $field_item) {
         foreach ($field_item as $column => $value) {
           if (!isset($properties['by_field'][$field_name][$column])) {
             continue;
           }
-          $pred = '<' . (string) $properties['by_field'][$field_name][$column] . '>';
-          if (!filter_var($value, FILTER_VALIDATE_URL) === FALSE) {
-            $obj = '<' . $value . '>';
+          // Skip the bundle as it is handled separately later.
+          if ($field_name == 'rid') {
+            continue;
+          }
+          // Don't add empty values.
+          if ($value === "") {
+            continue;
```

Comment:
Is there a specific reason for filtering out empty strings? When an entity is saved in the sql database, a field with an empty value will still be saved, right? 
Why wouldn't we save the empty string value? Because there is a difference between empty string and null.